deploy/keycloak/mcpserver-with-auth.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: toolhive.stacklok.dev/v1alpha1
+kind: MCPServer
+metadata:
+  name: fetch-server-keycloak
+  namespace: toolhive-system
+spec:
+  # Simple echo MCP server for testing
+  image: ghcr.io/stackloklabs/gofetch/server:0.0.4
+  resourceOverrides:
+    proxyDeployment:
+      env:
+        # by default we deploy KC w/o SSL
+        - name: INSECURE_DISABLE_URL_VALIDATION
```

Comment:
Ii just realized about this (probably undocumented) env var. Looks like we have a double negative naming concern here, people may misunderstand whether true means secure or insecure (what I actually did at first glance 😬 )